### PR TITLE
fix: allow env var templating for go2rtc rtsp creds

### LINKED ISSUE #5998: config: go2rtc rtsp to support the env variable template
**Describe what you are trying to accomplish and why in non technical terms**
In the config, have `go2rtc`'s rtsp section support environment variable swaps like in `streams`.

**Describe the solution you'd like**
Expand support for environment variable swaps in the config.
No more putting the raw password in the config file.

**Describe alternatives you've considered**
Currently, the raw password must be in the config for it to work. 

**Additional context**

[It's currently unsupported. Using curly braces takes the whole string is the literal password](https://github.com/blakeblackshear/frigate/issues/5982)

### PR DESCRIPTION
**Pull request by tnyeanderson**
Fixes: #5998 

Related to #5982

### CODE REVIEW

**Comment by tnyeanderson** [comment]
As a side question while I have the maintainer's attention, why not perform the environment variable substitution for the entire YAML string *first*, then send it through the YAML parser? This would allow variable interpolation anywhere in the config and allow a lot more flexibility for a lot less code.

File: docker/rootfs/usr/local/go2rtc/create_config.py [hunk 1]
```diff
@@ -64,8 +64,14 @@
...
-elif go2rtc_config["rtsp"].get("default_query") is None:
-    go2rtc_config["rtsp"]["default_query"] = "mp4"
+else:
+    if go2rtc_config["rtsp"].get("default_query") is None:
+        go2rtc_config["rtsp"]["default_query"] = "mp4"
+    # perform environment variable substitution for the go2rtc rtsp creds
+    if go2rtc_config["rtsp"].get("username") is not None:
+        go2rtc_config["rtsp"]["username"] = go2rtc_config["rtsp"]["username"].format(**FRIGATE_ENV_VARS)
+    if go2rtc_config["rtsp"].get("password") is not None:
+        go2rtc_config["rtsp"]["password"] = go2rtc_config["rtsp"]["password"].format(**FRIGATE_ENV_VARS)
```
Comment (NickM-27) [inline docker/rootfs/usr/local/go2rtc/create_config.py:74]:
```suggestion
    if go2rtc_config["rtsp"].get("default_query") is None:
        go2rtc_config["rtsp"]["default_query"] = "mp4"

    # perform environment variable substitution for the go2rtc rtsp creds

    if go2rtc_config["rtsp"].get("username") is not None:
        go2rtc_config["rtsp"]["username"] = go2rtc_config["rtsp"]["username"].format(**FRIGATE_ENV_VARS)

    if go2rtc_config["rtsp"].get("password") is not None:
        go2rtc_config["rtsp"]["password"] = go2rtc_config["rtsp"]["password"].format(**FRIGATE_ENV_VARS)
```

**Review by NickM-27 — CHANGES_REQUESTED** [review]

File: docker/rootfs/usr/local/go2rtc/create_config.py (hunk 1 above)
Comment (tnyeanderson) [reply docker/rootfs/usr/local/go2rtc/create_config.py]: This should be resolved now, added the two blank lines

**Comment by tnyeanderson** [comment]
I'm not sure I agree with the assessment above. The way it is currently implemented, it first checks for environment variables which are set and start with `FRIGATE_`. Then it replaces that brace-wrapped value in the resulting strings for each field where replacement is "allowed".

I don't think that there would be (m)any unintended consequences here. The user would have to set an env var with the `FRIGATE_` prefix *and* include an interpolation `{FRIGATE_MYVAR}` in the config file to have any effect.

>go2rtc makes use of {format} in many places throughout the go2rtc config

Are there any circumstances where `{FRIGATE_*}` format is used in the config file *and* where the env var has been set, but where the variable value should not be interpolated? I just can't imagine a scenario for this, but that's likely due to my unfamiliarity with the code and architecture.

The issue I see is that there is currently (at least) a few dozen lines of added complexity to perform the replacement on a per-field basis, and it is inflexible such that PRs like this have to exist for any place where the user might want to use interpolation. But should we even be telling the user what they can and can't do in this case? Maybe I want to control more options with env vars than the current setup allows, and every time that happens it has to be implemented (like in this PR), which is just more maintenance, tech debt, and complexity.

I'd argue it's also *less* easy to understand! I had to look at the source code to figure out which fields I'm "allowed" to use variable substitution in. Instead, the docs could say "if an environment variable exists and starts with `FRIGATE_`, use `{FRIGATE_MYVAR}` anywhere in the YAML config to template the variable's value". Isn't this easier to grasp, with less gotchas, then saying that it works in some places but not in others, and to search through issues and source code to find out why/where?

I don't mean this to be harsh at all. I love frigate and appreciate everyone's hard work on the project. I just want to make sure it is the best it can be, and that unneeded work can be minimized!

**Comment by blakeblackshear** [comment]
I can't remember why we decided to do them selectively at first. I don't see a great reason not to support it universally at this point.

**Comment by tnyeanderson** [comment]
>I can't remember why we decided to do them selectively at first. I don't see a great reason not to support it universally at this point.

Then I'll try to get a PR in for it this weekend, and the discussion can continue there.

Thanks for the amazing software @blakeblackshear !

**Comment by tnyeanderson** [comment]
Please reopen, this still should be merged in the meantime

**Comment by tnyeanderson** [comment]
Please reopen

**Comment by NickM-27** [comment]
You need to fix the errors in the checks

**Comment by tnyeanderson** [comment]
The `black` formatting should be fixed now. Please resolve the "changes requested" @NickM-27

File: docker/rootfs/usr/local/go2rtc/create_config.py [hunk 2]
```diff
@@ -64,8 +64,21 @@
...
 if go2rtc_config.get("rtsp") is None:
     go2rtc_config["rtsp"] = {"default_query": "mp4"}
-elif go2rtc_config["rtsp"].get("default_query") is None:
-    go2rtc_config["rtsp"]["default_query"] = "mp4"
+else:
+    if go2rtc_config["rtsp"].get("default_query") is None:
+        go2rtc_config["rtsp"]["default_query"] = "mp4"
+
+    # perform environment variable substitution for the go2rtc rtsp creds
+
```
Comment (NickM-27) [inline docker/rootfs/usr/local/go2rtc/create_config.py:72]:
```suggestion

```

**Review by NickM-27 — CHANGES_REQUESTED** [review]

File: docker/rootfs/usr/local/go2rtc/create_config.py (hunk 2 above)
Comment (NickM-27) [reply docker/rootfs/usr/local/go2rtc/create_config.py:72]: One more quick change then I think it should be good

Comment (tnyeanderson) [reply docker/rootfs/usr/local/go2rtc/create_config.py]: Done

**Review by NickM-27 — APPROVED** [review]

**Pull request merged**